runtime/server/catalog.go

```diff
@@ -5,15 +5,21 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/rilldata/rill/runtime/server/jwt"
+
 	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
 	"github.com/rilldata/rill/runtime/drivers"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
-	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
+	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
 // ListCatalogEntries implements RuntimeService.
 func (s *Server) ListCatalogEntries(ctx context.Context, req *runtimev1.ListCatalogEntriesRequest) (*runtimev1.ListCatalogEntriesResponse, error) {
+	err := authorizeInstanceID(ctx, req.InstanceId)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Would be great if we could have something like:
```go
claims := GetClaims(ctx)
if !claims.Reads(req.InstanceId) {
	// Return error
}
```

and if no auth info is found, `GetClaims` would still return a valid object corresponding either to full access (if local, i.e. no JWT secret) or no access (if cloud, i.e. a JWT secret is set)

Reply:
I will think some more about what permissions we will need in the JWT. Let me know if you have some thoughts already.

Reply:
Reads method check for the claim first with instance_id key and return true if its same as req.InstanceId